gulpfile.js

```diff
@@ -62,7 +66,24 @@ function generateScript(inFile, outFile, minify, callback, format) {
         .pipe(rename(outFile)),
   ];
   if (minify) {
-    parts.push(uglify());
+    if (format === 'es') {
+      parts.push(terser({
```

Comment:
@filipesilva is this the right configuration for terser esm2015 build? can you please double check. thanks

Reply:
Yeah, I am not sure about this `configuration part`, thanks.

Reply:
@JiaLiPassion, as a reference, the Terser settings we use in Angular CLI for applications can be found [here](https://github.com/angular/angular-cli/blob/58bc5b8f2a2a285f8e09ec05ef407fe32be3ef05/packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts#L228-L266).

I advise you to use only `terser` instead of `uglify` for some bundles and `terser` for other. At certain points in time either of these tools suffered from different hard to identify bugs. Ideally your bundles function the same. But if different formats use different minifiers, you may have different behaviour between them.

Under this mindset it is also preferable to keep as much of the minification settings identical between bundles.

Given that your existing es5 bundle uses default `uglify` settings, I recommend your es2015 bundle to use the same defaults and only pass in `{ ecma: 6 }`. This option will propagate to other Terser options automatically. I don't think you need any other options.

Reply:
@filipesilva, thank you very much for the help, currently, zone.js use `uglify` for `es5 bundle` and use `terser` for `es2015 bundle`, I will use `terser` to handle both `es5` and `es6`.
And I will also delete all other options.